### PR TITLE
Rename all from searx to searxng

### PR DESCRIPTION
**Pull request by HLFH**
## What does this PR do?

Rename all from searx to searxng (files, folder, content)

This change allows searxng to be a complete fork with its own name everywhere.

## Why is this change important?
This is important as searxng is now too different from searx with 600+ more commits. searxng is now a successful fork.

I was doing the AUR package for searxng on ArchLinux, and so before packaging searxng, it is best to rename everything to searxng.

### CODE REVIEW

**Comment by HLFH** [comment]
cc @dalf

**Comment by HLFH** [comment]
The AUR package [searxng-git](https://aur.archlinux.org/packages/searxng-git/) has been submitted. Not sure about the dependencies requirements.
I'd say it is a good start. In the meantime, the AUR package is pointing to my repo, happy to point to your searxng repo, once this PR gets merged.

File: searxng/data/external_bangs.json [hunk 1]
```diff
@@ -13686,8 +13686,8 @@
                         "outlet": "http://www.searsoutlet.com/d/products.jsp?md=srh_md&stxt=\u0002\u00010"
                     },
                     "rx": {
-                        "*": "//searx.me/?q=\u0002\u000170",
-                        ".me": "http://searx.me/?q=\u0002&pageno=1&category_general\u000118"
+                        "*": "//searxng.me/?q=\u0002\u000170",
+                        ".me": "http://searxng.me/?q=\u0002&pageno=1&category_general\u000118"
```
Comment (dalf) [inline searxng/data/external_bangs.json]: There are two DDG bangs `!searx` and `!searx.me`. 
There are no `searxng.me` domain.

File: searxng/flaskfix.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def __init__(self, wsgi_app):
             self.script_name = base_url.path
             if self.script_name.endswith('/'):
                 # remove trailing slash to avoid infinite redirect on the index
-                # see https://github.com/searx/searx/issues/2729
+                # see https://github.com/searxng/searxng/issues/2729
```
Comment (dalf) [inline searxng/flaskfix.py]: It should be `searx/searx`

**Comment by HLFH** [comment]
I can squash if needed ;)

**Comment by dalf** [comment]
Thank you !

It's huge.

File: docs/dev/plugins.rst [hunk 3]
```diff
@@ -35,13 +35,13 @@ External plugins
 
 External plugins are standard python modules implementing all the requirements of the standard plugins.
 Plugins can be enabled by adding them to :ref:`settings.yml`'s ``plugins`` section.
-Example external plugin can be found `here <https://github.com/asciimoo/searx_external_plugin_example>`_.
+Example external plugin can be found `here <https://github.com/asciimoo/searxng_external_plugin_example>`_.
```
Comment (dalf) [inline docs/dev/plugins.rst]: The actual repository is https://github.com/asciimoo/searx_external_plugin_example

File: docs/admin/installation.rst [hunk 4]
```diff
@@ -82,16 +82,16 @@ If all services are running fine, you can add it to your HTTP server:
...
-<https://www.thegreenwebfoundation.org/news/searching-the-green-web-with-searx/>`__:
+<https://www.thegreenwebfoundation.org/news/searching-the-green-web-with-searxng/>`__:
 
 .. code:: bash
 
-   $ sudo -H ./utils/searx.sh shell
+   $ sudo -H ./utils/searxng.sh shell
    // exit with [CTRL-D]
-   (searx-pyenv) searx@ryzen:~$ pip install git+https://github.com/return42/tgwf-searx-plugins
+   (searxng-pyenv) searxng@ryzen:~$ pip install git+https://github.com/return42/tgwf-searxng-plugins
```
Comment (dalf) [inline docs/admin/installation.rst]: No ng here

File: docs/admin/installation.rst [hunk 5]
```diff
@@ -82,16 +82,16 @@ If all services are running fine, you can add it to your HTTP server:
 
 Use SearXNG's ``shell`` to install external plugins.  In the example below we
 install the SearXNG plugins from **The Green Web Foundation** `[ref]
-<https://www.thegreenwebfoundation.org/news/searching-the-green-web-with-searx/>`__:
+<https://www.thegreenwebfoundation.org/news/searching-the-green-web-with-searxng/>`__:
```
Comment (dalf) [inline docs/admin/installation.rst]: No ng here

**Review by dalf — COMMENTED** [review]
Few comments.

The directory `docs/_themes/searx` must be renamed to `docs/_themes/searxng` (that's why the documentation doesn't compiled).

File: searxng/data/external_bangs.json (hunk 1 above)
Comment (HLFH) [reply searxng/data/external_bangs.json]: Fixed in local commit

File: searxng/flaskfix.py (hunk 2 above)
Comment (HLFH) [reply searxng/flaskfix.py]: Fixed in local commit

File: docs/dev/plugins.rst (hunk 3 above)
Comment (HLFH) [reply docs/dev/plugins.rst]: Fixed in local commit

File: docs/admin/installation.rst (hunk 4 above)
Comment (HLFH) [reply docs/admin/installation.rst]: link fixed in local commit

File: docs/admin/installation.rst (hunk 5 above)
Comment (HLFH) [reply docs/admin/installation.rst]: link fixed in local commit

**Comment by HLFH** [comment]
Doc directory and doc files with '*searx*' in it have been renamed to '*searxng*'.

**Comment by HLFH** [comment]
@dalf Thanks to your review, I think we can now merge ;)

**Comment by dalf** [comment]
Please hold on :-)
* I still haven't check on my laptop
* `make test` fail because of a code format issue (see https://github.com/searxng/searxng/pull/358/checks?check_run_id=3745893924#step:7:19 

https://github.com/searxng/searxng/blob/ac05f0943c6f7883a0b14d3b90af327dfe380e40/searxng/plugins/__init__.py#L29-L37
-->
```python
from searxng.plugins import (
    oa_doi_rewrite,
    ahmia_filter,
    hash_plugin,
    infinite_scroll,
    self_info,
    hostname_replace,
    search_on_category_select,
    tracker_url_remover,
    vim_hotkeys,
)
```

@return42 can you check the installation scripts ?

**Comment by return42** [comment]
> @return42 can you check the installation scripts ?

Yes, when I have time ...

This is a huge patch .. might break the history "here and there" .. should we really change all in one patch? .. if so, we should freeze other developments (PRs) and we need a really long test phase I think.

**Comment by return42** [comment]
This PR is on the master branch of @HLFH .. the question is, if we can push our fixes on top of this branch?

May be it is better we have this branch (with a different name, not master ;) at https://github.com/searxng/searxng so we (me and @dalf) can push our fixes on top ... will be much easier than explaining every detail in the github GUI.

**Comment by dalf** [comment]
To fix the tests ( `make test` ):
```patch
diff --git a/searxng/static/themes/simple/css/searxng.css b/searxng/static/themes/simple/css/searxng.css
index 580bf0ee..4366c284 100644
--- a/searxng/static/themes/simple/css/searxng.css
+++ b/searxng/static/themes/simple/css/searxng.css
@@ -1811,7 +1811,7 @@ td:hover .engine-tooltip,
   text-align: center;
 }
 .index .title {
-  background: url('../img/searxng.png') no-repeat;
+  background: url('../img/searx.png') no-repeat;
   width: 100%;
   min-height: 80px;
   background-position: center;
diff --git a/searxng/static/themes/simple/css/searxng.min.css b/searxng/static/themes/simple/css/searxng.min.css
index 85c95ee6..6d8c4fb0 100644
Binary files a/searxng/static/themes/simple/css/searxng.min.css and b/searxng/static/themes/simple/css/searxng.min.css differ
diff --git a/searxng/static/themes/simple/src/less/index.less b/searxng/static/themes/simple/src/less/index.less
index 9d396a6a..a74988a7 100644
--- a/searxng/static/themes/simple/src/less/index.less
+++ b/searxng/static/themes/simple/src/less/index.less
@@ -6,7 +6,7 @@
   text-align: center;
 
   .title {
-    background: url('../img/searxng.png') no-repeat;
+    background: url('../img/searx.png') no-repeat;
     width: 100%;
     min-height: 80px;
     background-position: center;
diff --git a/searxng/templates/simple/index.html b/searxng/templates/simple/index.html
index cddc5ee8..3e1359c6 100644
--- a/searxng/templates/simple/index.html
+++ b/searxng/templates/simple/index.html
@@ -1,7 +1,7 @@
 {% extends "simple/base.html" %}
 {% from 'simple/macros.html' import icon %}
 {% block meta %}
-	<link rel="preload" href="{{ url_for('static', filename='img/searxng.png') }}" as="image" />
+	<link rel="preload" href="{{ url_for('static', filename='img/searx.png') }}" as="image" />
 {% endblock %}
 {% block content %}
 <nav id="linkto_preferences"><a href="{{ url_for('preferences') }}">{{ icon('navicon-round') }}</a></nav>
diff --git a/tests/unit/test_webapp.py b/tests/unit/test_webapp.py
index 30626aeb..fb982df1 100644
--- a/tests/unit/test_webapp.py
+++ b/tests/unit/test_webapp.py
@@ -84,7 +84,7 @@ class ViewsTestCase(SearxTestCase):
         self.assertEqual(result.status_code, 200)
         self.assertIn(b'<div class="text-hide center-block" id="main-logo">'
                       + b'<img class="center-block img-responsive" src="/static/themes/oscar/img/logo_searx_a.png"'
-                      + b' alt="searxng logo" />searxng</div>', result.data)
+                      + b' alt="searx logo" />searxng</div>', result.data)
 
     def test_index_html_post(self):
         result = self.app.post('/', data={'q': 'test'})
```

It requires to modify the compiled CSS and JS files, which may break the css.map / js.map files.

There are other issue here and there: https://github.com/HLFH/searxng/blame/master/docs/index.rst#L9

---

To sum up, it requires to review all the changes, otherwise it will break things here and there.

I suggest to split this PR into different smaller PR focused on different components:

* phase 1, independent PR.
	* [ ] docker
		* create and update the searxng-docker project
		* `docs/*`
	* [ ] theme:common
		* `searx/templates/__common__`
	* [ ] theme:oscar
		* `searx/static/themes/oscar`
		* `searx/templates/themes/oscar`
		* build the theme
	* [ ] theme:simple
		* `searx/static/themes/simple`
		* `searx/templates/themes/simple`
		* build the theme
	* [ ] searx_extra
		* `searx_extra`
		* `.github`
		* `manage`
		* `docs/*` (where there is a reference to searx_extra)
	* [ ] setup (keep the searx module)
		* `setup.py`
		* `manage`
		* all the files in the root directory
		* `docs/*`
* phase 1 bis
	* [ ] SearXNG image(s) / icon(s)
		* update the themes, the documentations, etc...
* phase 2, the huge one
	* [ ] installation scripts
		* `utils/*.sh` (installation in searxng directory, but keep the searx module)
	* [ ] searx module
		* rename searx to searxng
		* take care of the translations branch / weblate
		* `docs/*`
		* `.github` workflows
		* `setup.py`
		* `manage`
		* all the files in the root directory
		* `docs/*`
* phase 3, clean up
	* [ ] everything else
		* git grep "searx" (especially the documentation)

A way to manage this:
* different issues with a common label (like `searxng migration`)
* a [project](https://github.com/searxng/searxng/projects)

It will take more times, but we can take time to review more carefully each PR while not stopping the development or require a double maintenance.

**Comment by return42** [comment]
> I suggest to split this PR into different smaller PR focused on different components:

I agree with that, in addition: when we move themes, we should move them directly to `/src`, see https://github.com/searxng/searxng/issues/141#issuecomment-890481065

**Comment by HLFH** [comment]
@dalf I am currently satisfied of my searxng instance working flawlessly. So I will currently keep my AUR searxng-git package as it is.

Since this PR will be split into multiple small PRs, I am closing this one.
And I definitely think searxng needs its own logo. Does not make sense to keep the searx logo.

**Comment by dalf** [comment]
> I am currently satisfied of my searxng instance working flawlessly. 

The code is huge boat, that's I've tried to stabilized. (I know this is subjective).
They are a lot different possible configurations / setups.

But clearly, this project needs to different name space.

> And I definitely think searxng needs its own logo.

Again, clearly, yes.
I've tried this https://gist.github.com/dalf/b670d030ce191c1a6a3826df5ac9366d but it could be better.... (https://github.com/searxng/searxng/issues/370 )

**Pull request closed without merging**